Error when running local integration tests using the latest nodeshift and openshift-rest-client

A new version of rhoaster was published and it is using the latest version of nodeshift and openshift-rest-client.

The previous rhoaster version `0.2.0` was pointing to the deprecated bucharest-gold image:

```console
15:31 $ npm run test:integration

> nodejs-rest-http@3.0.0 test:integration /home/hf/Documents/nodejs-rest-http
> tape test/integration/*.js | tap-spec

    No deployment found. Deploying...
    2019-10-24T18:31:51.700Z INFO loading configuration
    2019-10-24T18:31:51.701Z INFO using namespace myproject at https://192.168.99.101:8443
    2019-10-24T18:31:51.704Z INFO creating archive of package.json, app.js, public, bin, LICENSE, licenses
    2019-10-24T18:31:51.719Z INFO using existing build configuration nodejs-rest-http-s2i
    2019-10-24T18:31:51.724Z INFO using existing image stream nodejs-rest-http
    2019-10-24T18:31:51.724Z INFO uploading binary archive /home/hf/Documents/nodejs-rest-http/tmp/nodeshift/build/archive.tar
    2019-10-24T18:31:54.446Z INFO binary upload complete
    2019-10-24T18:31:54.446Z INFO waiting for build to finish
    2019-10-24T18:31:57.516Z TRACE Using bucharestgold/centos7-s2i-nodejs:latest as the s2i builder image
    2019-10-24T18:31:57.516Z TRACE ---> Installing application source
    2019-10-24T18:31:57.516Z TRACE ---> Building your Node application from source
    2019-10-24T18:31:57.516Z TRACE Current git config
```

So a new version was published to fix that, but an issue was found:

```console
$ npm run test:integration

> nodejs-rest-http@3.0.0 test:integration /home/hf/Documents/nodejs-rest-http
> tape test/integration/*.js | tap-spec

kubernetes-client deprecated require('kubernetes-client').config, use require('kubernetes-client/backends/request').config. node_modules/openshift-rest-client/lib/openshift-rest-client.js:25:17
kubernetes-client deprecated require('kubernetes-client').config, use require('kubernetes-client/backends/request').config. node_modules/openshift-rest-client/index.js:23:39
TypeError: openshift is not a function
    at Object.deploy (/home/hf/Documents/nodejs-rest-http/node_modules/rhoaster/index.js:40:12)


npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! nodejs-rest-http@3.0.0 test:integration: `tape test/integration/*.js | tap-spec`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the nodejs-rest-http@3.0.0 test:integration script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/hf/.npm/_logs/2019-10-24T18_45_17_405Z-debug.log
```

--- a/index.js
+++ b/index.js
@@ -1,7 +1,6 @@
 'use strict';
 
-const configLoader = require('openshift-config-loader');
-const openshift = require('openshift-rest-client');
+const { OpenshiftClient: restClient, config: k8sConfig } = require('openshift-rest-client');
 const nodeshift = require('nodeshift');
 const path = require('path');
 
@@ -33,21 +32,19 @@ function rhoaster (options) {
 
 function deploy (options) {
   return async () => {
-    const config = Object.assign(await configLoader(options), options);
+    const config = Object.assign(k8sConfig.fromKubeconfig(options.config), options);
     if (config.forceDeploy) {
       return runDeploy(config);
     }
-    return openshift({ config, request: { strictSSL: options.strictSSL } }).then(client => {
-      return client.deploymentconfigs.find(config.deploymentName)
-        .then(result => {
-          if (result.code === 404) throw new Error(result.reason);
-          console.log(`Deployment ${config.deploymentName} already exists. To force redeployment provide options.forceDeploy=true.`);
-          return applyResources(config);
-        })
-        .catch(_ => {
-          console.log('No deployment found. Deploying...');
-          return runDeploy(config).then(_ => waitForDeployment(client, config));
-        });
+    return restClient({ config }).then(async (client) => {
+      const result = await awaitRequest(client.apis['apps.openshift.io'].v1.ns(config.namespace).deploymentconfigs(config.deploymentName).get());
+      if (result.code === 404) {
+        console.log('No deployment found. Deploying...');
+        return runDeploy(config).then(_ => waitForDeployment(client, config));
+      }
+
+      console.log(`Deployment ${config.deploymentName} already exists. To force redeployment provide options.forceDeploy=true.`);
+      return applyResources(config);
     });
   };
 }
@@ -76,7 +73,7 @@ function runDeploy ({ projectLocation, strictSSL, dockerImage, nodeVersion }) {
 }
 
 function getRoute (output) {
-  return `http://${output.appliedResources.find(val => val.kind === 'Route').spec.host}`;
+  return `http://${output.appliedResources.find(val => val.body.kind === 'Route').body.spec.host}`;
 }
 
 function wait (timeout) {
@@ -85,11 +82,24 @@ function wait (timeout) {
   });
 }
 
+async function awaitRequest (promise) {
+  let result;
+  try {
+    result = await promise;
+  } catch (err) {
+    if (err.code !== 404) {
+      return Promise.reject(err);
+    }
+    result = err;
+  }
+  return result;
+}
+
 async function waitForDeployment (client, config, count) {
   count = count || 0;
   while (count++ < 50) {
-    const data = await client.deploymentconfigs.find(config.deploymentName);
-    if (data.status.availableReplicas > 0) {
+    const data = await awaitRequest(client.apis['apps.openshift.io'].v1.ns(config.namespace).deploymentconfigs(config.deploymentName).get());
+    if (data.body.status.availableReplicas > 0) {
       console.log('');
       return applyResources(config);
     } else {

--- a/package.json
+++ b/package.json
@@ -47,7 +47,6 @@
   },
   "dependencies": {
     "nodeshift": "~3.1.1",
-    "openshift-config-loader": "~0.4.0",
     "openshift-rest-client": "~2.3.0"
   }
 }

